frontend/src/components/common/ErrorModal.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react';
+import { Modal, Row, Col, Button, Container } from 'react-bootstrap';
+
+import { IGenericNetworkAction } from 'actions/genericActions';
+import { useHistory } from 'react-router-dom';
+
+const ErrorModal = (props: any) => {
```

Comment:
I like this! Wrapping all of our components into some sort of "error message" middleware/container.
However, the concept of "error boundaries" is kind of new in React. Could you please provide some JSDoc to help other devs understand the need for this? Thanks!